libraries/AP_Compass/AP_Compass_SITL.cpp

```diff
@@ -123,19 +134,19 @@ void AP_Compass_SITL::_timer()
         correct_field(f, _compass_instance[i]);
 
         _mag_accum[i] += f;
+        _accum_count[i]++;
     }
 
     if (!_sem->take(HAL_SEMAPHORE_BLOCK_FOREVER)) {
         return;
     }
 
-    _accum_count++;
-    if (_accum_count == 10) {
-        for (uint8_t i=0; i<SITL_NUM_COMPASSES; i++) {
+    for (uint8_t i=0; i<SITL_NUM_COMPASSES; i++) {
+        if (_accum_count[i] == 10) {
             _mag_accum[i] /= 2;
+            _accum_count[i] = 5;
+            _has_sample = true;
```

Comment:
_mag_accum[i] *= 0.5 for performance?
Also what is the significance of resetting each value to 5? Can this be changed to a descriptively-named variable? (trying to minimize introducing more magic numbers)

Reply:
this is a simulation change. It is a way of preventing overflow of accmulation counters. Normally accum_count stays small, if we stop reading samples for a while we need to prevent overflow.